source/AI.cpp

```diff
@@ -1978,7 +1980,21 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 	bool isJump = (ship.GetCheapestJumpType(ship.GetTargetSystem()).first == Ship::JumpType::JumpDrive);
 
 	Point direction = ship.GetTargetSystem()->Position() - ship.GetSystem()->Position();
-	if(!isJump && scramThreshold)
+	double squaredDeparture =  isJump ?
+		ship.GetSystem()->JumpDepartureDistance() * ship.GetSystem()->JumpDepartureDistance() :
+		ship.GetSystem()->HyperDepartureDistance() * ship.GetSystem()->HyperDepartureDistance();
+	squaredDeparture += SAFETY_OFFSET;
+	if(ship.Position().LengthSquared() < squaredDeparture)
+	{
+		Point closestDeparturePoint;
+		if(ship.Position().Length())
```

Comment:
`ship.Position()` can be translated to bool directly, no need to perform an expensive Pythagoras operation to get the `Length()` here.
```suggestion
		if(ship.Position())
```